packages/superset-ui-chart/src/query/Metric.ts

```diff
@@ -26,34 +26,37 @@ export enum Aggregate {
 }
 
 export enum ExpressionType {
+  BUILTIN = 'BUILTIN',
   SIMPLE = 'SIMPLE',
   SQL = 'SQL',
 }
 
-interface AdhocMetricSimple {
+interface MetricSimple {
```

Comment:
How about `SimpleMetric` and `SQLMetric`? But isn't everything through SQL? What does SQL Metric mean?

Reply:
So `SimpleMetric` is for the metrics that are generated only by an aggregation operator and a column. For `SqlMetric`, it is for a metric that with a SQL query specified. 

I don't have a strong opinion on `SimpleMetric` or `MetricSimple`.  either works for me 👍

Reply:
I like `SimpleMetric`, `SQLMetric` and `BuiltInMetric`. 👍